lego/apps/events/views.py

```diff
@@ -182,16 +179,27 @@ def payment(self, request, *args, **kwargs):
 
         if registration.has_paid():
             raise APIPaymentExists()
+
         registration.charge_status = constants.PAYMENT_PENDING
         registration.save()
-        chain(
-            async_payment.s(registration.id, serializer.data["token"]),
-            registration_payment_save.s(registration.id),
-        ).delay()
+        client_secret = (
```

Comment:
This makes ~no sense, as it will block the main thread to the celery worker completes, and we should _not_ do that. If that is the intention, it makes no sense to do it in celery instead of here directly. (ref. comment about rethinking the process)

Reply:
Also I don't understand the process here, as the mappings are related to the previous process.